### PR TITLE
Fix validation of RangeSlider

### PR DESCRIPTION
**Pull request by pberkes**
What I did in chronological order:
1) When editing the text field, the RangeSlider value should only update if the value is valid
2) Use a `TextField` instead of `LineEdit` to get feedback from the background color
3) Fix a problem with `TextField`: `valid` was not synchronized with `value` when `mode='auto'`

### CODE REVIEW

**Comment by pberkes** [comment]
Should be easier to review following the commits

File: qt_binder/tests/test_widgets.py [hunk 1]
```diff
@@ -56,8 +88,42 @@ def test_field_format_func(self):
...
+            range_slider.configure()
+            slider = range_slider.slider
+            field = range_slider.field
+
+            self.assertEqual(slider.value, 50)
+            self.assertEqual(range_slider.value, 50)
+
+            # State is Intermediate.
+            field.text = '1'
+            field.trait_property_changed('editingFinished', Undefined, True)
```
Comment (rkern) [inline qt_binder/tests/test_widgets.py]: Side question: is it worth wrapping up this pattern nicely somewhere? I assume that you did this to convince the `Binder` that the `editingFinished` event fired without actually triggering the Qt signal.

**Comment by rkern** [comment]
LGTM!

File: qt_binder/tests/test_widgets.py (hunk 1 above)
Comment (pberkes) [inline qt_binder/tests/test_widgets.py]: Right, imitating what out had in `test_field_text_edited` for `textEdited`:
`slider.field.trait_property_changed('textEdited', Undefined, 20)`.

Another way of doing this (better?) is
`field.qobj.editingFinished.emit()`

**Comment by pberkes** [comment]
In the tests that create Qt widgets I did not run the event loop, or did anything special to make sure that the widgets are disposed. I think it's ok but I don't know the `Binder` class well enough -- what do you think?

File: qt_binder/tests/test_widgets.py (hunk 1 above)
Comment (rkern) [inline qt_binder/tests/test_widgets.py]: I didn't have `GUITestAssistant` enabled at the time to let me actually `construct()` the objects with live Qt objects behind them. Now that you've added it, `field.editingFinished = True` should work.

Comment (pberkes) [inline qt_binder/tests/test_widgets.py]: or that :-D

**Comment by pberkes** [comment]
It worked well for `editingFinished`, but not for `textEdited`

**Comment by rkern** [comment]
Ugh, probably another PySide bug.

**Comment by rkern** [comment]
No, PyQt4 has the same problem.

**Comment by rkern** [comment]
You need to placate the `flake8`.

**Comment by pberkes** [comment]
Done

**Comment by pberkes** [comment]
Adapted following #40

File: qt_binder/tests/test_widgets.py [hunk 2]
```diff
@@ -14,13 +14,14 @@
 
 import unittest
 
-from traits.api import Undefined, pop_exception_handler, push_exception_handler
 from traits.testing.unittest_tools import UnittestTools
 
+from .test_raw_widgets import _BaseTestWithGui
```
Comment (rkern) [inline qt_binder/tests/test_widgets.py]: If we're going to use this in multiple places, maybe put it on its own somewhere?

Comment (rkern) [inline qt_binder/tests/test_widgets.py]: We can add a `@contextmanager` method to it too:

```
@contextmanager
def constructed(self, binder):
    binder.construct()
    try:
        binder.configure()
        yield binder
    finally:
        binder.dispose()


with self.constructed(TextField(mode='auto')) as field:
    field.textEdited = '10'
    ...
```

**Comment by rkern** [comment]
Refactoring the test utility can be done later. LGTM.

File: qt_binder/tests/test_widgets.py (hunk 2 above)
Comment (pberkes) [inline qt_binder/tests/test_widgets.py]: Sounds good, I'll move it out. How about `qt_binder.testing`?

Comment (rkern) [inline qt_binder/tests/test_widgets.py]: WFM.

**Comment by pberkes** [comment]
sorry I did  the refactoring here

**Comment by rkern** [comment]
LGTM!

**Pull request merged**